client_end_user/shared/src/commonMain/kotlin/presentation/screens/HomeViewModel.kt

```diff
@@ -0,0 +1,18 @@
+package presentation.screens
+
+
+import cafe.adriel.voyager.core.model.StateScreenModel
+import kotlinx.coroutines.flow.update
+import org.koin.core.component.KoinComponent
+
+class HomeViewModel : StateScreenModel<HomeUiState>(HomeUiState()), KoinComponent {
```

Comment:
There is actually no requirement to inherit the KoinComponent in this particular situation.
```suggestion
class HomeViewModel : StateScreenModel<HomeUiState>(HomeUiState()) {
```